### PR TITLE
Add test for #208

### LINKED ISSUE #208: Standard library function `panicking::take_hook` does not codegen
```
            // [compiler/rustc_codegen_llvm/src/gotoc/rvalue.rs:885] self.current_fn().readable_name() = "panicking::take_hook"
            // error: internal compiler error: compiler/rustc_middle/src/ty/fold.rs:829:17: Trying to collect bound vars with a bound region: DebruijnIndex(0) BoundRegion { var: 0, kind: BrAnon(0) }

```

related #109

### PR DESCRIPTION
**Pull request by zhassan-aws**
### Description of changes: 

Adding a regression for #208 which now works.

### Resolved issues:

Resolves #208 

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tests/expected/panic-hook/main.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+use std::panic;
+
+fn main() {
+    panic::set_hook(Box::new(|_| {
+        println!("Custom panic hook");
```
Comment (celinval) [inline tests/expected/panic-hook/main.rs:5]: Should we be checking for this?

Comment (zhassan-aws) [reply tests/expected/panic-hook/main.rs:5]: I tried to, but it didn't work. I'll file an issue and link it.

Comment (celinval) [reply tests/expected/panic-hook/main.rs]: Are you closing #208?

Comment (zhassan-aws) [reply tests/expected/panic-hook/main.rs]: Yes, this PR will close it.

Comment (celinval) [reply tests/expected/panic-hook/main.rs]: Makes sense to me.

**Comment by adpaco-aws** [comment]
I see #946 was opened but nothing happened here. @zhassan-aws can we close this?

**Comment by zhassan-aws** [comment]
@adpaco-aws, I still like to add the test to close #208, but I'm yet to address your comments. Will try to do that today.

**Comment by zhassan-aws** [comment]
It seems that `take_hook` is still not working (see https://github.com/model-checking/kani/issues/208#issuecomment-1083753409), so closing this PR.

**Pull request closed without merging**